src/main/resources/hudson/matrix/LabelAxis/label-axis-resources.js

```diff
@@ -0,0 +1,40 @@
+Behaviour.specify("DIV.labelAxis-tree", 'LabelAxis', 0, function(e) {
+    var tree = new YAHOO.widget.TreeView(e);
+
+    var i18nContainer = document.querySelector(".label-axis-i18n");
+    var labels = new YAHOO.widget.TextNode(i18nContainer.getAttribute("data-i18n-labels"), tree.getRoot(), false);
+    var machines = new YAHOO.widget.TextNode(i18nContainer.getAttribute("data-i18n-individual-nodes"), tree.getRoot(), false);
+
+    var values = (e.getAttribute("values") || "").split("/");
+    function has(v) {
+        return values.include(v) ? 'checked="checked" ' : "";
+    }
+
+    var labelAxisDataContainer = document.querySelector(".label-axis-data-container");
+    labelAxisDataContainer.childNodes.forEach(node => {
+        var labelCheckbox = node.getAttribute("data-label-checkbox");
+
+        // inserting 'checked' attribute after '<input ' hence index of 7
```

Comment:
An alternative to this comment is to do: `var afterInputIndex = '<input '.length` and then using that variable instead of the magical number.

Reply:
Sounds great, didn't think of it :D
Will adjust.